test/test_puma_server.rb

```diff
@@ -340,7 +340,7 @@ def test_early_hints_works
 
       HTTP/1.0 200 OK
       X-Hello: World
```

Comment:
This is an interesting one... should this be lowercased? 🤔

Reply:
IMO, this change will be unrelated to Puma. Whatever headers Puma is receiving, it's returning them in that format.

Since this has been already discussed in the issue thread https://github.com/puma/puma/issues/3250#issuecomment-1873550030. so we should update it to return it in lowercase.

But making these changes will increase the size of this pull request by more than 100 lines. Although the code changes will be minimal, but many test cases need to be updated. WDYT?

Reply:
> But making these changes will increase the size of this pull request by more than 100 lines.

What, me worry?

Most of the test changes will be single line changes, so I'll commit to reviewing them.  TIA if you decide to make the changes, and thanks for the work you've already done.

Reply:
ok, sure will do.